### ISSUE TITLE
Set Seed not achieving reproducible results

### ISSUE THREAD

**Issue by olivboom** [msc-acse/acse-9-independent-research-project-OliverJBoom#4]

**Comment by olivboom** [comment]
Can produce reproducible results on full dataset but not on batches. Will investigate further

**Comment by olivboom** [comment]
Can add set_seed in after each optimizer.zero_grad() call and it fixes it. Not sure why yet, something in optimizer.zero_grad() messes with the seeding.
